hplsql/src/main/java/org/apache/hive/hplsql/packages/PackageRegistry.java

```diff
@@ -0,0 +1,27 @@
+/*
+ *  Licensed to the Apache Software Foundation (ASF) under one
+ *  or more contributor license agreements.  See the NOTICE file
+ *  distributed with this work for additional information
+ *  regarding copyright ownership.  The ASF licenses this file
+ *  to you under the Apache License, Version 2.0 (the
+ *  "License"); you may not use this file except in compliance
+ *  with the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.apache.hive.hplsql.packages;
+
+import java.util.Optional;
+
+public interface PackageRegistry {
+  Optional<String> findPackage(String name);
```

Comment:
It communicates that the package might not exists. However the legacy code indeed uses nulls therefore eventually it converted to that. I think instead of removing the optional from here, eventually we should adjust the existing code to make use of the Optional instead of null. I just didn't want to do it as part of this jira.